### PR TITLE
Fixed issues with inline if (#252, #253 and #309)

### PR DESCRIPTION
**Pull request by lvcabral**

### CODE REVIEW

File: src/parser/Parser.ts [hunk 1]
```diff
@@ -933,7 +933,8 @@ export class Parser {
                     });
                 }
 
-                if (match(Lexeme.Else)) {
+                if (previous().kind !== Lexeme.Newline && match(Lexeme.Else)) {
+                    console.log("Else 2");
```
Comment (TwitchBronBron) [inline src/parser/Parser.ts]: Looks like you forgot to remove this `console.log` statement.

File: src/parser/Parser.ts [hunk 2]
```diff
@@ -911,7 +911,7 @@ export class Parser {
                 }
                 thenBranch = new Stmt.Block([thenStatement], peek().location);
 
-                while (match(Lexeme.ElseIf)) {
+                while (previous().kind !== Lexeme.Newline && match(Lexeme.ElseIf)) {
```
Comment (TwitchBronBron) [inline src/parser/Parser.ts]: There's already a `checkPrevious()` method that does this in a slightly less verbose way. @sjbarag might ask you to use that.

File: src/parser/Parser.ts (hunk 1 above)
Comment (lvcabral) [reply src/parser/Parser.ts]: indeed, will fix it.

File: src/parser/Parser.ts (hunk 2 above)
Comment (lvcabral) [reply src/parser/Parser.ts]: will take a look, thanks for the review

**Comment by sjbarag** [comment]
@lvcabral could you please run a `git rebase` against the upstream `master` branch and delete any commits that don't need to be here?  There's currently 526 commits and 61,000 lines added, but something tells me that's not what you meant to do :wink:

**Comment by lvcabral** [comment]
I'm closing this pull request, will resubmit with a new clean branch

**Pull request closed without merging**